crates/sc-consensus-subspace/src/block_import.rs

```diff
@@ -672,18 +683,39 @@ where
             }
         }
 
-        // The fork choice rule is that we pick the heaviest chain (i.e. smallest solution range),
-        // if there's a tie we go with the longest chain
+        // The fork choice rule is the largest fork solution range weight.
+        //
+        // This almost always prioritises:
+        // - the longest chain (the largest number of solutions), and if there is a tie
+        // - the strictest solutions (the numerically smallest solution ranges).
```

Comment:
Solutions do not matter. As you can see above, the calculation of added weight is only based on the solution range in the era, and not the quality (strictness in your terminology) of individual solutions. It was (incorrectly) based on quality of individual solutions long time ago during one of the testnets and it was susceptible to block withholding attacks and maybe something else too. So it is just the heaviest chain rule like Jeremy commented originally.